validator_client/src/doppelganger_service.rs

```diff
@@ -175,27 +174,26 @@ async fn beacon_node_liveness<'a, T: 'static + SlotClock, E: EthSpec>(
     } else {
         // Request the previous epoch liveness state from the beacon node.
         beacon_nodes
-            .first_success(
-                RequireSynced::Yes,
-                OfflineOnFailure::Yes,
-                |beacon_node| async {
-                    beacon_node
-                        .post_validator_liveness_epoch(previous_epoch, &validator_indices)
-                        .await
-                        .map_err(|e| format!("Failed query for validator liveness: {:?}", e))
-                        .map(|result| {
-                            result
-                                .data
-                                .into_iter()
-                                .map(|response| LivenessResponseData {
-                                    index: response.index,
-                                    epoch: previous_epoch,
-                                    is_live: response.is_live,
-                                })
-                                .collect()
-                        })
-                },
-            )
+            .first_success(|beacon_node| async {
+                let owned_beacon_node = beacon_node.clone();
+                drop(beacon_node);
```

Comment:
I am not particularly happy with this solution as it is not super clean

Reply:
How about this?

```rust
        beacon_nodes
            .first_success(|beacon_node| {
                let validator_indices_ref = &validator_indices;
                async move {
                    beacon_node
                        .post_validator_liveness_epoch(previous_epoch, &validator_indices_ref)
                        .await
...
```

Although I think we may not need an owned `beacon_node` passed to this closure?

Reply:
I think this has been resolved